tests/unit/src/test/scala/tests/debug/StepDapSuite.scala

```diff
@@ -0,0 +1,172 @@
+package tests.debug
+
+import tests.BaseDapSuite
+import scala.meta.internal.metals.debug.DebugStep._
+import scala.meta.internal.metals.debug.DebugFileLayout
+import scala.meta.internal.metals.debug.StepNavigator
+
+object StepDapSuite extends BaseDapSuite("debug-step") {
+  assertSteps("step-out")(
+    sources = List(
+      """|a/src/main/scala/Main.scala
+         |package a
+         |
+         |object Main {
+         |  def foo() = {
+         |>>  println()
+         |  }
+         |
+         |  def main(args: Array[String]): Unit = {
+         |    foo()
+         |  }
+         |}
+         |""".stripMargin
+    ),
+    main = "a.Main",
+    instrument = steps =>
+      steps
+        .at("a/src/main/scala/Main.scala", line = 5)(StepOut)
+        .at("a/src/main/scala/Main.scala", line = 9)(Continue)
```

Comment:
Could we avoid hardcoding the line numbers here somehow? For example, using comment markers 
```scala
             |>>  println() // debug: step 1
             
// ...
           |    foo() // debug: step 2
// ...
                .at("a/src/main/scala/Main.scala", line = "step 1")(StepOut)
                .at("a/src/main/scala/Main.scala", line = "step 2")(Continue)
```

Reply:
That would be problematic for cases when a single line is hit multiple times. Also, developer would have to either declare the methods in order of breakpoints or somehow indicate the order within the commit markers which I think would be less readable than having all of the steps in one place. Also, we would need to encode the step type in the comment and it all becomes not worth the hassle in my opinion. 
The test cases are small enough that they shouldn't be changed that often, so let's hope we won't have too many reasons to do so.

Reply:
Fair enough, as a general principle I try to avoid hardcoded line numbers in test cases because the line numbers change when you change the code.